### ISSUE TITLE
[Enum] `unused_field` fails with `_index`

### ISSUE THREAD

**Issue by bernaferrari** [dart-lang/linter#4247]
Check the following example:

```dart
enum MyEnum {
  first(1),
  second(2);

  // works
  const MyEnum(this.value);
  final int value;

  // fails
  const MyEnum(this._index);
  final int _index; // unused_field
}
```

Probably `_index` might be clashing with `Enum.index`.

**Comment by srawlins** [comment]
Hmm, I think this is expected. The field is unused.

**Comment by bernaferrari** [comment]
But it doesn't happen with `value` which is also kind of unused, right?

**Comment by srawlins** [comment]
Since `value` is public, we cannot tell if it is used (in another library) or not.

**Comment by bernaferrari** [comment]
But since _index is used in the constructor, it is been used, right?

**Comment by lrhn** [comment]
It's not being *used*. It's being written, but never read, which is the definition of being "unused".
If you remove the variable, *and the write to it* (since you can't write to something which isn't there), the program keeps working, so the variable isn't *necessary*.

**Comment by bernaferrari** [comment]
I don't understand, but I'm going to close.
